### PR TITLE
[TT-1413] fixed tracing, more complex debug contract, DOT graph generation

### PR DESCRIPTION
**Pull request by Tofel**
* fixed missing nested calls in tracing and added recursive processing
* missing 4byte or opcode tracing won't fail the process, just print warn
* errors when decoding inputs/outputs/events won't fail tracing, just print warn
* prettied console tracing
* added DOT graph generation
* added CLI to trace single transaction hash passed as cmd line flag
* updated tracing tests to include `ParentSignature`, `NestingLevel` and `CallType`
* updated README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -342,6 +349,16 @@ avg_tps = 3.0
 max_tps = 8.0
 ```
 
+### Single transaction tracing
+You can trace a single transaction using `seth trace` command. Example:
+```
+SETH_CONFIG_PATH=seth.toml go run cmd/seth/seth.go -u "https://my-rpc.network.io" trace -t 0x4c21294bf4c0a19de16e0fca74e1ea1687ba96c3cab64f6fca5640fb7b84df65
```
Comment (skudasov) [inline README.md:355]: Can we use `seth` alias here?

File: README.md [hunk 2]
```diff
@@ -482,16 +500,37 @@ It's important to know that in order to use congestion metrics we need to fetch
...
 For both transaction types if any of the steps fails, we fallback to hardcoded values.
 
+### DOT graphs
+There are multiple ways of visualising DOT graphs:
+* `xdot` application [recommended]
+* VSCode Extensions
+* online viewers
+
+### xdot
+To install simply run `hombrew install xdot` and then run `xdot <path_to_dot_file>`. This tool seems to be the best for the job, since the viewer is interactive and supports tooltips, which in our case contain extra tracing information.
```
Comment (skudasov) [inline README.md:510]: typo, homebrew

Comment (skudasov) [reply README.md:510]: Also, consider `nix` https://search.nixos.org/packages?channel=unstable&show=xdot&from=0&size=50&sort=relevance&type=packages&query=xdot

File: dot_graph.go [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+				newPath = append(newPath, call.CommonData.Signature)
+				queue = append(queue, node{call: call, path: newPath})
+			}
+		}
+	}
+
+	return nil // No path found
+}
+
+var defaultTruncateTo = 20
```
Comment (skudasov) [inline dot_graph.go:78]: Is that the maximum height of the tree, should it be a parameter?

**Review by skudasov — CHANGES_REQUESTED** [review]

File: dot_graph.go (hunk 3 above)
Comment (Tofel) [reply dot_graph.go:78]: no, that's for truncating input values of functions as these sometimes are `[]uint` of varying length, but they easily can have > 100 elements and then the graph becomes really ugly and harder to read

File: README.md (hunk 2 above)
Comment (Tofel) [reply README.md:510]: Not sure it makes sense as `nix develops` errors, when trying to install `xvfb`:
```
error: Package ‘xvfb-run-1+g87f6705’ in /nix/store/byybjfdwhkpv18xqj7945ca755gbisc3-source/pkgs/tools/misc/xvfb-run/default.nix:59 is not available on the requested hostPlatform:
         hostPlatform.config = "aarch64-apple-darwin"
         package.meta.platforms = [
           "aarch64-linux"
           "armv5tel-linux"
           "armv6l-linux"
           "armv7a-linux"
           "armv7l-linux"
           "i686-linux"
           "loongarch64-linux"
           "m68k-linux"
           "microblaze-linux"
           "microblazeel-linux"
           "mips-linux"
           "mips64-linux"
           "mips64el-linux"
           "mipsel-linux"
           "powerpc64-linux"
           "powerpc64le-linux"
           "riscv32-linux"
           "riscv64-linux"
           "s390-linux"
           "s390x-linux"
           "x86_64-linux"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.
```

**Review by skudasov — APPROVED** [review]

**Pull request merged**